pkg/pool-weighted/contracts/lbp/LBPool.sol

```diff
@@ -254,11 +248,37 @@ contract LBPool is WeightedPool, Ownable2Step, BaseHooks {
         return IRouterCommon(router).getSender() == owner();
     }
 
+    /**
```

Comment:
Also, how serious are we about making the sale *completely* deterministic? It is still possible for the owner to change the price or sell more tokens by adding liquidity during the sale (a feature of the original LBP, though rarely used).

If we want it to be completely predictable, we could block adding liquidity after the sale starts. (You might then have a dead contract if you failed to initialize it before the start of the sale. Ok I guess as long as funds can't be locked. Worst case, they have to wait for the end of the "sale" to withdraw again.)